test/common/router/router_test.cc

```diff
@@ -868,6 +872,121 @@ TEST_F(RouterTest, EnvoyUpstreamServiceTime) {
   EXPECT_TRUE(verifyHostUpstreamStats(1, 0));
 }
 
+// Validate that x-envoy-attempt-count is added to request headers when option is true.
+TEST_F(RouterTest, EnvoyAttemptCountInRequest) {
+  setIncludeAttemptCount(true);
```

Comment:
optional: do you want to name this to Request even though it doesn't reflect the config option until v4?

Reply:
I wanted to keep the name change in #10325. So I will update this or that other PR depending on merge order.